rcl/test/rcl/test_time.cpp

```diff
@@ -263,11 +263,13 @@ TEST(CLASSNAME(rcl_time, RMW_IMPLEMENTATION), specific_clock_instantiation) {
   rcl_allocator_t allocator = rcl_get_default_allocator();
   {
     rcl_clock_t uninitialized_clock;
-    rcl_ret_t ret = rcl_clock_init(
-      RCL_CLOCK_UNINITIALIZED, &uninitialized_clock, &allocator);
+    rcl_ret_t ret = rcl_clock_init(RCL_CLOCK_UNINITIALIZED, &uninitialized_clock, &allocator);
     EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
     EXPECT_EQ(uninitialized_clock.type, RCL_CLOCK_UNINITIALIZED) <<
       "Expected time source of type RCL_CLOCK_UNINITIALIZED";
+    ret = rcl_clock_fini(&uninitialized_clock);
+    EXPECT_EQ(ret, RCL_RET_INVALID_ARGUMENT) << rcl_get_error_string().str;
+    rcl_reset_error();
```

Comment:
This is ok, but it still seems unrelated to the PR.
You can add a `RCUTILS_CHECK_ALLOCATOR_IS_VALID` check after initializing the clock, to test this modification.

Reply:
Changed in `[ce8d867](https://github.com/ros2/rcl/commit/ce8d867187741775b3bb22da355bb16acbf3d42a)`